demos/smart_framing_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -0,0 +1,287 @@
+#include "custom_kernels.hpp"
+
+#include <algorithm>
+
+#include <opencv2/gapi/cpu/gcpukernel.hpp>
+
+#include <opencv2/gapi/imgproc.hpp>
+#include <opencv2/gapi/core.hpp>
+#include <opencv2/gapi/operators.hpp>
+
+namespace {
+
+struct YOLOv4TinyPostProcessing {
+
+    int region_num = 3;
+    int region_classes = 80;
+    int region_coords = 4;
+    //YOLO v4
+    int netInputHeight = 416;
+    int netInputWidth = 416;
+
+    double intersectionOverUnion(const custom::DetectedObject& o1, const custom::DetectedObject& o2) {
+        double overlappingWidth = fmin(o1.x + o1.width, o2.x + o2.width) - fmax(o1.x, o2.x);
+        double overlappingHeight = fmin(o1.y + o1.height, o2.y + o2.height) - fmax(o1.y, o2.y);
+        double intersectionArea = (overlappingWidth < 0 || overlappingHeight < 0) ? 0 : overlappingHeight * overlappingWidth;
+        double unionArea = o1.width * o1.height + o2.width * o2.height - intersectionArea;
+        return intersectionArea / unionArea;
+    }
+
+    std::vector<float> defaultAnchors = {
+        // YOLOv4_Tiny
+        { 10.0f, 14.0f, 23.0f, 27.0f, 37.0f, 58.0f,
+          81.0f, 82.0f, 135.0f, 169.0f, 344.0f, 319.0f}
+    };
+
+    std::vector<int64_t> defaultMasks = {
+        // YOLOv4_Tiny
+        {1, 2, 3, 3, 4, 5}
+    };
+
+    static inline float sigmoid(float x) {
+        return 1.f / (1.f + exp(-x));
+    }
+    static inline float linear(float x) {
+        return x;
+    }
+
+    int calculateEntryIndex(int totalCells, int lcoords, int lclasses, int location, int entry) {
+        int n = location / totalCells;
+        int loc = location % totalCells;
+        return (n * (lcoords + lclasses + 1) + entry) * totalCells + loc;
+    }
+
+    std::string getLabelName(int labelID) { return (size_t)labelID < custom::coco_classes.size() ? custom::coco_classes[labelID] : std::string("Label #") + std::to_string(labelID); }
+
+    std::vector<float> blobs_anchor[2]{
+        {23.0f, 27.0f, 37.0f, 58.0f, 81.0f, 82.0f},
+        {81.0f, 82.0f, 135.0f, 169.0f, 344.0f, 319.0f}
+    };
+
+    void parseYOLOOutput(
+        const cv::Mat& blob, int blobID, const unsigned long resized_im_h,
+        const unsigned long resized_im_w, const unsigned long original_im_h,
+        const unsigned long original_im_w, const float confidenceThreshold,
+        std::vector<custom::DetectedObject>& objects) {
+        // --------------------------- Extracting layer parameters -------------------------------------
+        int sideW = 0;
+        int sideH = 0;
+        unsigned long scaleH;
+        unsigned long scaleW;
+        //case YOLO_V3:
+        //case YOLO_V4:
+        //case YOLO_V4_TINY:
+        sideH = static_cast<int>(blob.size[2]);
+        sideW = static_cast<int>(blob.size[3]);
+        std::cout << "parseYOLOOutput  sideH " << sideH << " sideW " << sideW << std::endl;
```

Comment:
slog::debug << ...

and for the rest couts

Reply:
Done